{"uuid":"80ffca33-ba47-4dd5-9dff-ef71cf0d88df","historyId":"daf5885c172b30a0b031191d879bf7a4","fullName":"com.qacart.shaft.Api.sendRequest","labels":[{"name":"package","value":"com.qacart.shaft.Api"},{"name":"testClass","value":"com.qacart.shaft.Api"},{"name":"testMethod","value":"sendRequest"},{"name":"parentSuite","value":"Suite"},{"name":"suite","value":"shaft-demo"},{"name":"subSuite","value":"com.qacart.shaft.Api"},{"name":"host","value":"TDV-HQ-L21"},{"name":"thread","value":"10260@TDV-HQ-L21.main(1)"},{"name":"framework","value":"testng"},{"name":"language","value":"java"}],"links":[],"name":"sendRequest","status":"failed","statusDetails":{"known":false,"muted":false,"flaky":false,"message":"Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\". Root cause: \"java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\"","trace":"java.lang.AssertionError: Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\". Root cause: \"java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\"\r\n\tat org.testng.Assert.fail(Assert.java:99)\r\n\tat com.shaft.tools.io.internal.FailureReporter.fail(FailureReporter.java:34)\r\n\tat com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n\tat com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n\tat com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:303)\r\n\tat com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n\tat com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n\tat org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n\tat org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n\tat org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n\tat org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n\tat org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n\tat java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n\tat org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n\tat org.testng.TestRunner.run(TestRunner.java:621)\r\n\tat org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n\tat org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n\tat org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n\tat org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n\tat org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n\tat org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n\tat org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n\tat org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n\tat org.testng.TestNG.runSuites(TestNG.java:1114)\r\n\tat org.testng.TestNG.run(TestNG.java:1082)\r\n\tat com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n\tat com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)\r\nCaused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\r\n\tat com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n\tat com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n\tat com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:299)\r\n\t... 27 more\r\n"},"stage":"finished","steps":[{"name":"Assertion Failed: Evaluating that the response is successful (Status code is between 200 and 299)...","status":"passed","stage":"finished","steps":[],"attachments":[{"name":"Validation Test Data - Failure Reason","source":"0ee849e5-7280-444a-80d0-daa0d46b4423-attachment"}],"parameters":[{"name":"customLog","value":"Assertion Failed: Evaluating that the response is successful (Status code is between 200 and 299)..."},{"name":"attachments","value":"[[Validation Test Data, Failure Reason, java.lang.AssertionError: Assertion \"ValidateEquals\" FAILED. Expected \"true\" and Actual \"false\". Comparison Type \"EQUALS\". Validation Type \"POSITIVE\".\r\n com.shaft.validation.internal.ValidationsHelper.reportValidationState(ValidationsHelper.java:639)\r\n com.shaft.validation.internal.ValidationsHelper.fail(ValidationsHelper.java:395)\r\n com.shaft.validation.internal.ValidationsHelper.validateEquals(ValidationsHelper.java:70)\r\n com.shaft.validation.internal.ValidationsExecutor.perform(ValidationsExecutor.java:144)\r\n com.shaft.api.RestActions.evaluateResponseStatusCode(RestActions.java:1223)\r\n com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:293)\r\n com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n org.testng.TestRunner.run(TestRunner.java:621)\r\n org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n org.testng.TestNG.runSuites(TestNG.java:1114)\r\n org.testng.TestNG.run(TestNG.java:1082)\r\n com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)\r\n]]"}],"start":1688931741265,"stop":1688931741273},{"name":"Assertion \"ValidateEquals\" FAILED. Expected \"true\" and Actual \"false\". Comparison Type \"EQUALS\". Validation Type \"POSITIVE\".","status":"failed","stage":"finished","steps":[],"attachments":[],"parameters":[],"start":1688931741275,"stop":1688931741275},{"name":"Perform request failed \"POST | Target Status Code: 0 | Service URL: https://todo.qacart.com/api/v1/users/login | Content Type: application/json | Response Time: 1519ms\".","status":"passed","statusDetails":{"known":false,"muted":false,"flaky":false,"trace":"Headers:\r\n_______________\r\n\r\nContent-Type=application/json\r\njava.io.ByteArrayInputStream@5ec80bbd\r\njava.io.ByteArrayInputStream@30f74965"},"stage":"finished","steps":[],"attachments":[{"name":"API Request - Specifications","source":"7e103b5a-a2c6-47be-ba8c-eb540c063e98-attachment"},{"name":"API Request - JSON Body","source":"072c6b2f-1fd5-45ba-ba53-74c023353618-attachment.json","type":"text/json"},{"name":"API Response - JSON Body","source":"6c247c0e-ea7c-4b78-acd8-9a477c1ce882-attachment.json","type":"text/json"}],"parameters":[{"name":"logText","value":"Perform request failed \"POST | Target Status Code: 0 | Service URL: https://todo.qacart.com/api/v1/users/login | Content Type: application/json | Response Time: 1519ms\"."},{"name":"attachments","value":"[[API Request, Specifications, Headers:\r\n_______________\r\n\r\nContent-Type=application/json\r\n\r\n], [API Request, JSON Body, java.io.ByteArrayInputStream@5ec80bbd], null, [API Response, JSON Body, java.io.ByteArrayInputStream@30f74965]]"},{"name":"status","value":"FAIL"}],"start":1688931741904,"stop":1688931742080},{"name":"Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\".","status":"passed","statusDetails":{"known":false,"muted":false,"flaky":false,"trace":"Headers:\r\n_______________\r\n\r\nContent-Type=application/json\r\njava.io.ByteArrayInputStream@5c936b5f\r\njava.io.ByteArrayInputStream@5c2fc86\r\njava.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\r\n com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:299)\r\n com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n org.testng.TestRunner.run(TestRunner.java:621)\r\n org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n org.testng.TestNG.runSuites(TestNG.java:1114)\r\n org.testng.TestNG.run(TestNG.java:1082)\r\n com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)"},"stage":"finished","steps":[],"attachments":[{"name":"API Request - Specifications","source":"4473353b-e902-4b4d-a940-4e5f840cd7fd-attachment"},{"name":"API Request - JSON Body","source":"65f9ec28-cc8f-47db-903e-fdb153f65713-attachment.json","type":"text/json"},{"name":"API Response - JSON Body","source":"5f9d5a21-fc39-4e62-9a9d-b0a63c5209f9-attachment.json","type":"text/json"},{"name":"API Action Exception - Perform request - Stacktrace","source":"6e75009d-fe47-4e75-8bad-44825b454424-attachment"}],"parameters":[{"name":"logText","value":"Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\"."},{"name":"attachments","value":"[[API Request, Specifications, Headers:\r\n_______________\r\n\r\nContent-Type=application/json\r\n\r\n], [API Request, JSON Body, java.io.ByteArrayInputStream@5c936b5f], null, [API Response, JSON Body, java.io.ByteArrayInputStream@5c2fc86], [API Action Exception - Perform request, Stacktrace, java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\r\n com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:299)\r\n com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n org.testng.TestRunner.run(TestRunner.java:621)\r\n org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n org.testng.TestNG.runSuites(TestNG.java:1114)\r\n org.testng.TestNG.run(TestNG.java:1082)\r\n com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)\r\n]]"},{"name":"status","value":"FAIL"}],"start":1688931742082,"stop":1688931742099},{"name":"Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\". Root cause: \"java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\"","status":"passed","statusDetails":{"known":false,"muted":false,"flaky":false,"trace":"java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\r\n com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:299)\r\n com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n org.testng.TestRunner.run(TestRunner.java:621)\r\n org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n org.testng.TestNG.runSuites(TestNG.java:1114)\r\n org.testng.TestNG.run(TestNG.java:1082)\r\n com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)"},"stage":"finished","steps":[],"attachments":[{"name":"Rest actions - Get  stack  trace - Exception Stacktrace","source":"f00115a8-723b-43ce-84ba-a9359ae97055-attachment"}],"parameters":[{"name":"logText","value":"Perform request failed \"https://todo.qacart.com/api/v1/users/login, Response Time: 1519ms\". Root cause: \"java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\""},{"name":"attachments","value":"[[Rest actions - Get  stack  trace, Exception Stacktrace, java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0\r\n com.shaft.api.RestActions.failAction(RestActions.java:100)\r\n com.shaft.api.RestActions.failAction(RestActions.java:459)\r\n com.shaft.api.RequestBuilder.performRequest(RequestBuilder.java:299)\r\n com.shaft.api.RequestBuilder.perform(RequestBuilder.java:264)\r\n com.qacart.shaft.Api.sendRequest(Api.java:18)\r\n java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\r\n java.base/java.lang.reflect.Method.invoke(Method.java:578)\r\n org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)\r\n org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)\r\n org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)\r\n org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)\r\n org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)\r\n org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)\r\n org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)\r\n java.base/java.util.ArrayList.forEach(ArrayList.java:1511)\r\n org.testng.TestRunner.privateRun(TestRunner.java:848)\r\n org.testng.TestRunner.run(TestRunner.java:621)\r\n org.testng.SuiteRunner.runTest(SuiteRunner.java:443)\r\n org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)\r\n org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)\r\n org.testng.SuiteRunner.run(SuiteRunner.java:336)\r\n org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)\r\n org.testng.TestNG.runSuitesLocally(TestNG.java:1200)\r\n org.testng.TestNG.runSuites(TestNG.java:1114)\r\n org.testng.TestNG.run(TestNG.java:1082)\r\n com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)\r\n com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)\r\n]]"},{"name":"status","value":"FAIL"}],"start":1688931742105,"stop":1688931742107}],"attachments":[{"name":"SHAFT Engine Logs - Current Method log: sendRequest","source":"692a7d9d-f0be-4c2a-ad8d-d4afa190c48a-attachment.txt","type":"text/plain"}],"parameters":[],"start":1688931738358,"stop":1688931742123}